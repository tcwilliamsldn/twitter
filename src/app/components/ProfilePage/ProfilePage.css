.profilepage {
    @media (min-width: 768px) {
        display: flex;
        align-items: start;
        max-width: 1080px;
        margin: 0 auto;
    }
}
