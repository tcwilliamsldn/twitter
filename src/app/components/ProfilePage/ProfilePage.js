import React, { Fragment } from 'react';
import { Query } from 'react-apollo';
import Profile from './Profile/Profile';
import Feed from './Feed/Feed';
import Loading from '../Loading/Loading';
import Error from '../Error/Error';
import getUser from '../../queries/getUser';

export const ProfilePage = ({ loading, error, data }) => {
    if (loading) return <Loading />;
    if (error) return <Error message={error.message} />;

    const { tweets, ...restProps } = data.twitter.user;

    return (
        <Fragment>
            <Profile {...restProps} following={120} />
            <Feed tweets={tweets} />
        </Fragment>
    );
};

const ProfilePageWithData = ({ match }) => (
    <Query query={getUser(match.params.profileName)}>{ProfilePage}</Query>
);

export default ProfilePageWithData;
