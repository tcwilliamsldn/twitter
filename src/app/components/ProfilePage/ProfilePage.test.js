import { ProfilePage } from './ProfilePage';
import React from 'react';
import { shallow } from 'enzyme';

describe('ProfilePage', function() {
    it('renders the loading state when loading', function() {
        const component = shallow(<ProfilePage loading />);
        expect(component).toMatchSnapshot();
    });

    it('renders the error state on error', function() {
        const component = shallow(
            <ProfilePage error={{ message: 'Whoops' }} />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders a user Profile and Feed on success', function() {
        const data = { twitter: { user: { tweets: [], name: 'Tom' } } };

        const component = shallow(<ProfilePage data={data} />);
        expect(component).toMatchSnapshot();
    });
});
