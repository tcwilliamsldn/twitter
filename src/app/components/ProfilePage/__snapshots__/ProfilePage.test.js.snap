// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage renders a user Profile and Feed on success 1`] = `
<React.Fragment>
  <Profile
    following={120}
    name="Tom"
  />
  <Feed
    tweets={Array []}
  />
</React.Fragment>
`;

exports[`ProfilePage renders the error state on error 1`] = `
<Error
  message="Whoops"
/>
`;

exports[`ProfilePage renders the loading state when loading 1`] = `<Loading />`;
